source/Halibut/Transport/RewindableBufferStream.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.IO;
+
+namespace Halibut.Transport
+{
+    class RewindableBufferStream : Stream, IRewindableBuffer
+    {
+        readonly Stream baseStream;
+        readonly byte[] rewindBuffer;
+        bool rewindBufferPopulated;
+        int rewindBufferOffset;
+        int rewindBufferCount;
+        bool rewindEnabled;
+
+        public RewindableBufferStream(Stream baseStream, int rewindBufferSize = 8192)
```

Comment:
Out of interest was this size picked as it was the default [buffer size in DeflateStream](https://github.com/microsoft/referencesource/blob/master/System/sys/system/IO/compression/DeflateStream.cs#L18)?

Reply:
Indeed, the buffer size was picked because it's the unconfigurable size of the `DeflateStream` buffer. Does Halibut have an existing configuration mechanism to hook into? Otherwise, it looks like the only option would be passing an arg down from the top?

Reply:
There `RewindableBufferStreamSize` value is now configurable.